trio/_ssl.py

```diff
@@ -207,15 +214,17 @@ class NeedHandshakeError(Exception):
 
     """
 
+    __slots__ = ()
+
```

Comment:
Geh this should probably also move out. I won't make this review "requested changes" though :P

Reply:
Removed in cf7a777